azurerm/resource_arm_network_security_rule.go

```diff
@@ -112,15 +112,19 @@ func resourceArmNetworkSecurityRule() *schema.Resource {
 				ConflictsWith: []string{"destination_address_prefix"},
 			},
 
+			//lintignore:S018
 			"source_application_security_group_ids": {
+				// todo: check? but it appears we only take one despite the API allowing multiple
 				Type:     schema.TypeSet,
 				MaxItems: 1,
 				Optional: true,
 				Elem:     &schema.Schema{Type: schema.TypeString},
 				Set:      schema.HashString,
 			},
 
+			//lintignore:S018
 			"destination_application_security_group_ids": {
+				// todo: check? but it appears we only take one despite the API allowing multiple
```

Comment:
Should we be renaming this to `source_application_security_group_id` string then?

Reply:
(same here) - I think it's worth leaving this matching the API since it's likely this'll support multiple at some point, given the way this is designed